rpc/src/v1/impls/eth_signing.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Eth Signing RPC implementation.
+
+use std::sync::{Arc, Weak};
+use jsonrpc_core::*;
+use ethminer::MinerService;
+use ethcore::client::BlockChainClient;
+use util::numbers::*;
+use util::keys::store::AccountProvider;
+use v1::helpers::{SigningQueue, ConfirmationsQueue};
+use v1::traits::EthSigning;
+use v1::types::TransactionRequest;
+use v1::impls::sign_and_dispatch;
+
+
+/// Implementation of functions that require signing when no trusted signer is used.
+pub struct EthSigningQueueClient {
+	queue: Weak<ConfirmationsQueue>,
+}
+
+impl EthSigningQueueClient {
+	/// Creates a new signing queue client given shared signing queue.
+	pub fn new(queue: &Arc<ConfirmationsQueue>) -> Self {
+		EthSigningQueueClient {
+			queue: Arc::downgrade(queue),
+		}
+	}
+}
+
+impl EthSigning for EthSigningQueueClient  {
+
+	fn sign(&self, _params: Params) -> Result<Value, Error> {
+		// TODO [ToDr] Implement sign when rest of the signing queue is ready.
+		rpc_unimplemented!()
+	}
+
+	fn send_transaction(&self, params: Params) -> Result<Value, Error> {
+		from_params::<(TransactionRequest, )>(params)
+			.and_then(|(request, )| {
+				let queue = take_weak!(self.queue);
+				queue.add_request(request);
+				// TODO [ToDr] Block and wait for confirmation?
```

Comment:
Returning the zero hash here is an acceptable option for the time being given that the spec seems to allow it, and especially since there is a fix pending in another branch.